napari/layers/points/points.py

```diff
@@ -2111,7 +2116,7 @@ def to_mask(
         )
         radii_scale = point_data_to_world_scale * mask_world_to_data_scale
 
-        output_data_radii = mean_radii * np.atleast_2d(radii_scale)
+        output_data_radii = np.atleast_2d(radii) * np.atleast_2d(radii_scale)
```

Comment:
I think the reason this change breaks the failing test is because `output_data_radii` is transposed in that case and because of the behavior of `zip` with 2D arrays.

The goal here is to multiply an nx1 (per-point size) by a 1x1 or 1xd (isotropic or per-dimension scale) to get an nxd, so this should work since `radii` is always of shape (N,) now.

```suggestion
        output_data_radii = radii[:, np.newaxis] * np.atleast_2d(radii_scale)
```

Reply:
Appreciate that some renaming or additional implementation comments could help here, so feel free to make those if you want.